server/routes/events.js

```diff
@@ -0,0 +1,98 @@
+var express = require('express')
+var passport = require("passport")
+var router = express.Router()
+var schoolSchema = require('../models/schoolSchema')
+var alumniSchema = require('../models/alumniSchema')
+var userSchema = require('../models/userSchema')
+var eventSchema = require('../models/eventSchema')
+require('mongoose').Promise = global.Promise
+
+
+/*
+    Helper function to ensure user has access to a given school
+    @param accessContexts - list of access levels for the given user
+    @param userSchool - user school object
+    @param school - object of school user is attempting to access
+    @return Boolean determining if the user has access to school
+*/
+const isValidAccessContext = ({ accessContexts, userSchool, school }) => {
+    if (accessContexts.includes("GLOBAL")) {
+        return true
+    } else if (accessContexts.includes("INTERSCHOOL")) {
+        return userSchool.country === school.country
+    } else {
+        return userSchool._id.toString() === school._id.toString()
+    }
+}
+
+
+/*
+    API used by alumni to create alumni accessible events
+    @return the user created event
+*/
+router.post('/create/:userId', async (req, res, next) => {
```

Comment:
We should guard this API by `passport.authenticate`

Reply:
Gooooooood catch. I forgot to up-date that post adding tests.

Reply:
Oh one thought, I don't think we should submit the userId with the request, I think we should be able to just figure that out from the request details of the logged in user? Noticed this while building the front end part

Reply:
@sbetti02 This may cue in a broader backend refactor for how we handle role-based access and authentication. Good eye. We can discuss this during the next standup.